reana_workflow_engine_serial/tasks.py

```diff
@@ -92,16 +89,49 @@ def run_serial_workflow(workflow_uuid, workflow_workspace,
         for command in step['commands']:
             current_command_idx += 1
             job_spec = {
-                'experiment': os.getenv('REANA_WORKFLOW_ENGINE_EXPERIMENT',
-                                        'serial_experiment'),
-                'docker_img': step['environment'],
-                'cmd': 'bash -c \"cd {0} ; {1} \"'.format(
+                "experiment": os.getenv("REANA_WORKFLOW_ENGINE_EXPERIMENT",
+                                        "serial_experiment"),
+                "docker_img": step["environment"],
+                "cmd": "bash -c \"cd {0} ; {1} \"".format(
                     workflow_workspace, escape_shell_arg(command)),
-                'max_restart_count': 0,
-                'env_vars': {},
-                'job_type': 'kubernetes',
-                'shared_file_system': True,
+                "max_restart_count": 0,
+                "env_vars": {},
+                "job_type": "kubernetes",
+                "shared_file_system": True,
             }
+            job_spec_copy = dict(job_spec)
+            clean_cmd = job_spec_copy['cmd'].split(';')[1]
+            job_spec_copy['cmd'] = clean_cmd
+            _, http_response = rjc_api_client.job_cache.check_if_cached(
+                job_spec=json.dumps(job_spec_copy),
+                workflow_json=json.dumps(step),
+                workflow_workspace=workflow_workspace).\
+                result()
+            result = http_response.json()
+            if result['cached']:
+                os.system('cp -R {source} {dest}'.format(
+                    source=os.path.join(result['result_path'],
+                                        'workspace', '*'),
+                    dest=workflow_workspace))
+                print('~~~~~ Copied from cache')
+                last_step = step
+                job_id = result['job_id']
+                if step == workflow_json['steps'][-1] and \
```

Comment:
TODO: This is copied from below, to publish the progress status when a job result is restored from the cache, but should be refactored to avoid duplicate code.